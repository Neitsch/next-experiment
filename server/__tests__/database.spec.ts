import { createConnection } from "typeorm";

import db from "../database";

jest.unmock("../database");

describe("Database", () => {
  it("dev", async () => {
    db({ dev: true })();
    // @ts-ignore
    expect(createConnection.mock.calls[0][0]).toMatchSnapshot();
  });
  it("prod", async () => {
    db({ dev: false })();
    // @ts-ignore
    expect(createConnection.mock.calls[0][0]).toMatchSnapshot();
  });
});
