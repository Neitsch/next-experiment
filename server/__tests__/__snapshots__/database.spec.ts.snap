// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Database dev 1`] = `
Object {
  "database": "db.sqlite",
  "entities": Array [
    "server/entity/account/*.ts",
  ],
  "migrations": Array [
    "server/migration/**/*.ts",
  ],
  "subscribers": Array [
    "server/subscriber/**/*.ts",
  ],
  "synchronize": true,
  "type": "sqlite",
}
`;

exports[`Database prod 1`] = `
Object {
  "database": "test",
  "entities": Array [
    "server/entity/account/*.ts",
  ],
  "host": "localhost",
  "migrations": Array [
    "server/migration/**/*.ts",
  ],
  "password": "test",
  "port": 5432,
  "subscribers": Array [
    "server/subscriber/**/*.ts",
  ],
  "synchronize": true,
  "type": "postgres",
  "username": "test",
}
`;
